src/Symfony/Bundle/FrameworkBundle/Controller/ControllerTrait.php

```diff
@@ -326,6 +326,28 @@ protected function createFormBuilder($data = null, array $options = array()): Fo
         return $this->container->get('form.factory')->createBuilder(FormType::class, $data, $options);
     }
 
+    /**
+     * Handles request and check form validity.
+     *
+     * @final
+     */
+    protected function isFormValid(FormInterface $form, Request $request = null): bool
+    {
+        if ($form->isSubmitted()) {
+            throw new \LogicException('The form is already submitted, use $form->isValid() directly.');
+        }
+
+        if (!$request) {
+            $request = $this->container->get('request_stack')->getCurrentRequest();
```

Comment:
Maybe we should check that the request stack service is defined to throw a better exception?

Reply:
I have copied this code from other method. 
IMHO this is fine as this